x/isIn.js

```diff
@@ -0,0 +1,146 @@
+const includes = require('./includes')
+const curry2 = require('../_internal/curry2')
+const __ = require('../_internal/placeholder')
+
+/**
+ * @name _isIn
+ *
+ * @synopsis
+ * ```coffeescript [specscript]
+ * _isIn(value any, container Array|Object|String|Set|Map) -> boolean
+ * ```
+ *
+ * @description
+ * Counterpart to includes. Check if a collection includes another value.
+ *
+ * ```javascript [playground]
+ * import isIn from 'https://unpkg.com/rubico/dist/x/isIn.es.js'
+ *
+ * console.log(
+ *   isIn(1, [1, 2, 3])
+ * ) // true
+ *
+ * console.log(
+ *   isIn(4, [1, 2, 3])
+ * ) // false
+ *
+ * console.log(
+ *   isIn(1, { a: 1 })
+ * ) // true
+ *
+ * console.log(
+ *   isIn(2, { a: 1 })
+ * ) // true
+ *
+ * console.log(
+ *   isIn('a', 'abc')
+ * ) // true
+ *
+ * console.log(
+ *   isIn('ab', 'abc')
+ * ) // true
+ *
+ * console.log(
+ *   isIn('d', 'abc')
+ * ) // false
+ *
+ * console.log(
+ *   isIn(1, new Set([1, 2, 3]))
+ * ) // true
+ *
+ * console.log(
+ *   isIn(4, new Set([1, 2, 3]))
+ * ) // false
+ *
+ * console.log(
+ *   isIn(1, new Map([[1, 1], [2, 2], [3, 3]]))
+ * ) // true
+ *
+ * console.log(
+ *   isIn(4, new Map([[1, 1], [2, 2], [3, 3]]))
+ * ) // false
+ * ```
+ */
+const _isIn = function (value, container) {
+  if (!container) {
+    return false
+  }
+  if (container.constructor == Set) {
+    return container.has(value)
+  }
+  if (container.constructor == Map) {
+    return Array.from(container.values()).includes(value)
+  }
+
+  return includes(value)(container)
```

Comment:
the only other data structures that we need are Array and String, both of which should have the includes property function. I would have a check if (typeof container.includes == 'function') {...} and just return false by default

Reply:
but what about Object? i use includes for Array/String and Object

Reply:
@richytong

Reply:
ah that's right, includes handles Array and String and Object